### DISCUSSION TITLE
Update iOS on mobile data without user action?

### DISCUSSION THREAD

**Discussion by MLB-BIOMETRIC** [micromdm/micromdm#988] [Q&A]
I am currently on the outlook for a new MDM system, because I have major issues with my current one when it comes to updating the iOS of the devices.

Currently, our problem is that **we're not able to do an iOS Software Update while the device is on mobile data without the user allows it on the device**. 

**Does anyone know if its possible without the user action?** We want the update to happen in the background. Furthermore, we don't care about restarts or similar - we just want the iOS to be updated without the user being involved.

I have taken a look at https://developer.apple.com/documentation/devicemanagement/scheduleosupdatecommand/command/updatesitem?changes=latest_minor
and maybe installAction ``InstallForceRestart`` or ``InstallASAP`` could work?

If it is a hard restriction from Apple, can anyone think of another method that we can simulate WiFi such that Apple updates?

**Comment by jessepeterson** [comment]
Good question; I'm not sure the answer. I have a feeling it is an Apple limitation (I've seen this popup with another commercial MDM vendor).

Have you tried asking in the #mdm or #mdmdev channels on the [MacAdmins Slack](https://www.macadmins.org)  or maybe the Apple Developer Forums? This likely isn't a MicroMDM support issue per se, at best its an Apple MDM configuration issue (if it is even something that can be configured).

**Reply by MLB-BIOMETRIC** [reply]
Thanks for you reply @jessepeterson !

Yea, i think this might be an Apple limitation as well... Really annoying when your devices are on mobile data and without user action.

Thanks for suggestion to look at Apple MDM configuration. I tried looking in the Profile Configurator, but doesn't seem to have any settings regarding updates (other than defer updates for X days).

I just feel there must be a way to update autonomously over mobile data, but maybe im wrong. 😄

**Reply by petarov** [reply]
It is a very good question to which I also do not have an answer, but definitely a customer that I expect to get to that same question at some point.

I guess it probably makes sense to open a feature request in the Feedback Assistant. The more the merrier.

**Reply by MLB-BIOMETRIC** [reply]
@petarov Yea, i created feature request ticket to the Apple Configurator support... Let's see if something happens :D
